beacon-chain/db/kv/state.go

```diff
@@ -367,40 +365,47 @@ func (s *Store) HasState(ctx context.Context, blockRoot [32]byte) bool {
 	return hasState
 }
 
+// checkJustifiedAndFinalized check if blockRoot is finalized, genesis or head
+func checkJustifiedAndFinalized(ctx context.Context, tx *bolt.Tx, blockRoot [32]byte) error {
```

Comment:
Unless there is a reuse of this logic happening in the context of this PR, I would prefer this logic to be left untouched